rom/laghos.cpp

```diff
@@ -1315,130 +1337,209 @@ int main(int argc, char *argv[])
+                        dmd_V = new CAROM::NonuniformDMD(outputPath + "/" + "dmdV" + romOptions.basisIdentifier + "_" + to_string(curr_window));
+                        dmd_E = new CAROM::NonuniformDMD(outputPath + "/" + "dmdE" + romOptions.basisIdentifier + "_" + to_string(curr_window));
+                    }
+                    else
+                    {
+                        dmd_X = new CAROM::DMD(outputPath + "/" + "dmdX" + romOptions.basisIdentifier + "_" + to_string(curr_window));
+                        dmd_V = new CAROM::DMD(outputPath + "/" + "dmdV" + romOptions.basisIdentifier + "_" + to_string(curr_window));
+                        dmd_E = new CAROM::DMD(outputPath + "/" + "dmdE" + romOptions.basisIdentifier + "_" + to_string(curr_window));
+                    }
 
-                    if (visit)
+                    prev_window = curr_window;
+                    window_start_time = curr_time;
+
+                    // Skip first sample of each subsequent window except the first.
+                    if (curr_window != 0)
                     {
-                        oper->ComputeDensity(*rho_gf);
-                        visit_dc->SetCycle(ti);
-                        visit_dc->SetTime(t);
-                        visit_dc->Save();
+                        continue;
                     }
                 }
+
+                if (result_X != NULL) delete result_X;
+                if (result_V != NULL) delete result_V;
+                if (result_E != NULL) delete result_E;
+
+                if (myid == 0) cout << "Predicting time t " << curr_time << " using DMD window " << curr_window << " with initial start time " << window_start_time << std::endl;
+
+                result_X = dmd_X->predict(curr_time);
+                result_V = (romOptions.useVX && romOptions.dmd_nonuniform) ? dmd_X->predict(curr_time, 1) : dmd_V->predict(curr_time);
+                result_E = dmd_E->predict(curr_time);
+                Vector m_result_X(result_X->getData(), result_X->dim());
+                Vector m_result_V(result_V->getData(), result_V->dim());
+                Vector m_result_E(result_E->getData(), result_E->dim());
+                x_gf->SetFromTrueDofs(m_result_X);
+                v_gf->SetFromTrueDofs(m_result_V);
+                e_gf->SetFromTrueDofs(m_result_E);
+
+                if (visit)
+                {
+                    oper->ComputeDensity(*rho_gf);
+                    visit_dc->SetCycle(ti);
+                    visit_dc->SetTime(t);
+                    visit_dc->Save();
+                }
             }
-            else
+            restoreTimer.Stop();
+        }
+        else
+        {
+            // -restore phase
+            // No need to specify t_final because the loop in -restore phase is determined by the files in ROMsol folder.
+            // When -romhr or --romhr are used in -online phase, then -restore phase needs to be called to project rom solution back to FOM size
```

Comment:
Long lines. Has astyle been run? Did it not catch this?

Reply:
make style is just run in the last commit. It changes few lines but not these ones. In fact, there are even longer ones, e.g. line 423.